### PR TITLE
*: replace our own metrics with codahale/metrics

### PR DESCRIPTION
**Pull request by xiang90**

### CODE REVIEW

File: rafthttp/entry_writer.go [hunk 1]
```diff
@@ -71,12 +65,12 @@ func (ew *entryWriter) writeEntry(ent *raftpb.Entry) error {
...
 	return err
 }
 
 func (ew *entryWriter) stop() {
-	metrics.GetMap("rafthttp.stream.entries_sent").Delete(ew.id.String())
-	metrics.GetMap("rafthttp.stream.bytes_sent").Delete(ew.id.String())
-	metrics.GetMap("rafthttp.stream.last_index_sent").Delete(ew.id.String())
+	metrics.Counter("rafthttp.stream.bytes_sent." + ew.id.String()).Remove()
+	metrics.Counter("rafthttp.stream.entries_sent." + ew.id.String()).Remove()
+	metrics.Counter("rafthttp.stream.last_index_sent." + ew.id.String()).Remove()
```
Comment (yichengq) [inline rafthttp/entry_writer.go]: is this gauge?

File: rafthttp/entry_reader.go [hunk 2]
```diff
@@ -67,12 +61,12 @@ func (er *entryReader) readEntry(ent *raftpb.Entry) error {
...
 	return ent.Unmarshal(buf)
 }
 
 func (er *entryReader) stop() {
-	metrics.GetMap("rafthttp.stream.entries_received").Delete(er.id.String())
-	metrics.GetMap("rafthttp.stream.bytes_received").Delete(er.id.String())
-	metrics.GetMap("rafthttp.stream.last_index_received").Delete(er.id.String())
+	metrics.Counter("rafthttp.stream.bytes_received." + er.id.String()).Remove()
+	metrics.Counter("rafthttp.stream.entries_received." + er.id.String()).Remove()
+	metrics.Counter("rafthttp.stream.last_index_received." + er.id.String()).Remove()
```
Comment (yichengq) [inline rafthttp/entry_reader.go]: is this gauge?

**Comment by yichengq** [comment]
lgtm after address the comments

**Comment by xiang90** [comment]
@yichengq fixed. merging.

**Pull request merged**